crates/starknet-server/tests/test_dump_and_load.rs

```diff
@@ -217,4 +247,171 @@ mod dump_and_load_tests {
+    }
+
+    #[tokio::test]
+    async fn dump_load_endpoints_transaction_and_state_after_load_is_valid() {
+        let dump_file_name = "dump_endpoint";
+        let dump_file_name_custom_path = "dump_endpoint_custom_path";
+        let devnet_dump =
+            BackgroundDevnet::spawn_with_additional_args(&["--dump-path", dump_file_name])
+                .await
+                .expect("Could not start Devnet");
+        let mint_tx_hash = devnet_dump.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+        let dump_body = Body::from(
+            json!({
+                "path": ""
+            })
+            .to_string(),
+        );
+        devnet_dump.post_json("/dump".into(), dump_body).await.unwrap();
+        assert!(Path::new(dump_file_name).exists());
+        let dump_body_custom_path = Body::from(
+            json!({
+                "path": dump_file_name_custom_path
+            })
+            .to_string(),
+        );
+        devnet_dump.post_json("/dump".into(), dump_body_custom_path).await.unwrap();
+        assert!(Path::new(dump_file_name_custom_path).exists());
+
+        let devnet_load = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let load_body = Body::from(
+            json!({
+                "path": dump_file_name
+            })
+            .to_string(),
+        );
+        devnet_load.post_json("/load".into(), load_body).await.unwrap();
+        let entry_point_selector =
+            starknet_rs_core::utils::get_selector_from_name("balanceOf").unwrap();
+        let balance_result = devnet_load
+            .json_rpc_client
+            .call(
+                FunctionCall {
+                    contract_address: FieldElement::from_hex_be(ERC20_CONTRACT_ADDRESS).unwrap(),
+                    entry_point_selector,
+                    calldata: vec![DUMMY_ADDRESS.into()],
+                },
+                BlockId::Tag(BlockTag::Latest),
+            )
+            .await
+            .expect("Failed to call contract");
+        assert_eq!(balance_result[0], DUMMY_AMOUNT.into());
+
+        let loaded_transaction =
+            devnet_load.json_rpc_client.get_transaction_by_hash(mint_tx_hash).await.unwrap();
+        if let starknet_rs_core::types::Transaction::Invoke(
+            starknet_rs_core::types::InvokeTransaction::V1(invoke_v1),
+        ) = loaded_transaction
+        {
+            assert_eq!(invoke_v1.transaction_hash, mint_tx_hash);
+        } else {
+            panic!("Could not unpack the transaction from {loaded_transaction:?}");
+        }
+
+        if Path::new(dump_file_name).exists() {
+            remove_file(dump_file_name);
+        } else {
+            panic!("Could not find dump file {}", dump_file_name);
+        }
+        if Path::new(dump_file_name_custom_path).exists() {
+            remove_file(dump_file_name_custom_path);
+        } else {
+            panic!("Could not find dump file {}", dump_file_name_custom_path);
+        }
+    }
```

Comment:
Why another round of existence assertion? Wasn't it already checked at this point?

Reply:
Fixed in 63e664ee8429af41b9ca690d52682f0f2dae2f31